internal/app/tfsec/checks/oci001.go

```diff
@@ -0,0 +1,66 @@
+package checks
+
+import (
+	"fmt"
+
+	"github.com/tfsec/tfsec/internal/app/tfsec/parser"
+	"github.com/tfsec/tfsec/internal/app/tfsec/scanner"
+)
+
+const OCIComputeIpReservation scanner.RuleCode = "OCI001"
+const OCIComputeIpReservationDescription scanner.RuleSummary = "Compute instance requests an IP reservation from a public pool"
+const OCIComputeIpReservationImpact = "The compute instance has the ability to be reached from outside"
+const OCIComputeIpReservationResolution = "Reconsider the use of an public IP"
+const OCIComputeIpReservationExplanation = `
+Compute instance requests an IP reservation from a public pool
+
+The compute instance has the ability to be reached from outside, you might want to sonder the use of a non public IP.
+`
+const OCIComputeIpReservationBadExample = `
+resource "opc_compute_ip_address_reservation" "my-ip-address" {
+	name            = "my-ip-address"
+	ip_address_pool = "public-ippool"
+  }
+`
+const OCIComputeIpReservationGoodExample = `
+resource "opc_compute_ip_address_reservation" "my-ip-address" {
+	name            = "my-ip-address"
+	ip_address_pool = "cloud-ippool"
+  }
+`
+
+func init() {
+	scanner.RegisterCheck(scanner.Check{
+		Code: OCIComputeIpReservation,
+		Documentation: scanner.CheckDocumentation{
+			Summary:     OCIComputeIpReservationDescription,
+			Explanation: OCIComputeIpReservationExplanation,
+			Impact:      OCIComputeIpReservationImpact,
+			Resolution:  OCIComputeIpReservationResolution,
+			BadExample:  OCIComputeIpReservationBadExample,
+			GoodExample: OCIComputeIpReservationGoodExample,
+			Links: []string{
+				"https://registry.terraform.io/providers/hashicorp/opc/latest/docs/resources/opc_compute_ip_address_reservation",
+				"https://registry.terraform.io/providers/hashicorp/opc/latest/docs/resources/opc_compute_instance",
+			},
+		},
+		Provider:       scanner.OracleProvider,
+		RequiredTypes:  []string{"resource"},
+		RequiredLabels: []string{"opc_compute_ip_address_reservation"},
+		CheckFunc: func(check *scanner.Check, block *parser.Block, _ *scanner.Context) []scanner.Result {
+			if attr := block.GetAttribute("ip_address_pool"); attr != nil {
+				if attr.IsAny("public-ippool") {
```

Comment:
If there is only one value with is an issue you can use `attr.Equals("public-ippool", parser.IngnoreCase)`

Reply:
Found the testing pieces... I will start writing one... 

Please do have some patience with me, it is a learning curve which we need to go through. As soon as we have nailed this I think we can get some people within my team who can leverage the learning curve I am taking right ow and help adding new checks ;)

Reply:
Hey @louwersj no pressure at all, really appreciate you helping out

Reply:
Added the first attempt to write a test for the OCI001 check...

Reply:
Hey @louwersj  - looking at the docs, it seems like `ip_address_pool` attribute is better as an `Equals`